### ISSUE TITLE
Angular module federation host app does not change LOCALE_ID when using @angular/localize

### ISSUE THREAD

**Issue by minijus** [nrwl/nx#13690]
### Current Behavior

Angular module federation host app does not change `LOCALE_ID` when using `@angular/localize`.
When using MF host for Angular app `LOCALE_ID` always stays as `sourceLocale` `en-US`.
This causes issues with locale dependant pipes (i.e. date, currency).

### Expected Behavior

Angular module federation host app should properly build localized versions of application where `LOCALE_ID` is changed to respective locale.

### Github Repo

https://github.com/minijus/localize-repro

### Steps to Reproduce

1. Clone repo https://github.com/minijus/localize-repro
2. `npm ci`
3. Follow `README.md`

OR

1. Generate new Angular host application `npx nx g @nrwl/angular:host`
2. Add i18n setup
3. Build with production configuration `npx nx run app-name:build`
4. Serve prod build `npx http-server dist/apps/app-name/locale`

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 18.12.1
   OS   : darwin arm64
   npm  : 9.1.3
   
   nx : 15.3.0
   @nrwl/angular : 15.3.0
   @nrwl/cypress : 15.3.0
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.3.0
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.3.0
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.3.0
   @nrwl/js : 15.3.0
   @nrwl/linter : 15.3.0
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/webpack : 15.3.0
   @nrwl/workspace : 15.3.0
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

### Failure Logs

```shell
There are no failure/error logs.
```

**Comment by minijus** [comment]
Should be fixed with https://github.com/angular/angular-cli/pull/24404

Thank you @Coly010 and @alan-agius4 !

**Comment by Coly010** [comment]
@minijus  This was fixed in Angular 15.0.4, so if you bump your `@angular-devkit/*` packages and `@angular/*` packages in your package.json to `~15.0.4` it should be resolved :)
